### ISSUE TITLE
Closing and reopening a maximized IDE doesn't return it to its original position when unmaximized

### ISSUE THREAD

**Issue by ronblum** [rstudio/rstudio#12992]
### System details

    RStudio Edition : Desktop
    RStudio Version : 2022.05.0-daily+212
    OS Version      : MacOS 13.2
    R Version       : 4.2.3

### Steps to reproduce the problem

1. Open Rstudio Desktop.
2. Move the IDE so that it's not in the center of the screen.
3. Double-click on the title bar to maximize (fills the screen).
4. Exit RStudio.
5. Open RStudio.
    - RStudio is maximized (fills the screen).
6. Double-click on the title bar to unmaximize it (return it to its previous size). 
    - The IDE isn't in the previous location.
    - Instead, it's positioned with the left edge or the right edge aligned with the center of the screen, depending on whether it was previously positioned more towards the right or left side of the screen.

### Describe the problem in detail

The IDE doesn't return to its last position.

### Describe the behavior you expected

The IDE should return to its last position.

Note: This originated when verifying Windows issue #12463.

- [ ] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [ ] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [ ] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by gtritchie** [comment]
Here's one way to reproduce the window disappear on restore ("unmaximize").

* Launch RStudio
* double-click title bar to maximize
* minimize the window (yellow circle in title bar)
* redisplay the window by clicking on it in dock
* quit and restart RStudio (it should start maximized)
* double-click title bar to unmaximize

The wIndow shrinks to nothing. If you quit then restart the window will reappear (albeit in a very tiny size).

**Comment by gtritchie** [comment]
This seems like an Electron bug (or perhaps a macOS bug). I'm seeing similar weirdness with VSCode if I use the native Window Title Bar Style (which is what RStudio uses). I logged an issue for that: https://github.com/microsoft/vscode/issues/180138

**Comment by gtritchie** [comment]
At a minimum we should ensure we never restore the window to an invisible size/location.

**Comment by kevinushey** [comment]
Is this something we could detect and recover from?

**Comment by gtritchie** [comment]
Yes, but seems worth waiting a bit to see if this gets addressed in macOS or perhaps Electron. Curious what (if anything) VSCode decides to do about it.
